pkgs/by-name/ht/htb-toolkit/package.nix

```diff
@@ -0,0 +1,73 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, openssl
+, stdenv
+, darwin
+, coreutils
+, gnome
+, libsecret
+, bash
+, openvpn
+, nerdfonts
+, gzip
+, killall
+}:
+
+rustPlatform.buildRustPackage {
+  pname = "htb-toolkit";
+  version = "unstable-2024-01-17";
+
+  src = fetchFromGitHub {
+    owner = "D3vil0p3r";
+    repo = "htb-toolkit";
+    # https://github.com/D3vil0p3r/htb-toolkit/issues/3
+    rev = "54e11774ea8746ea540548082d3b25c22306b4fc";
+    hash = "sha256-QYUqdqFV9Qn+VbJTnz5hx5I0XV1nrzCoCKtRS7jBLsE=";
+  };
+
+  cargoHash = "sha256-XDE6A6EIAUbuzt8Zb/ROfDAPp0ZyN0WQ4D1gWHjRVhg=";
+
+  # Patches only relevant for Nixpkgs
+  patches = [
+    ./main.patch
+    ./manage.patch
+    ./play.patch
+    ./types.patch
+    ./utils.patch
```

Comment:
I think we should give those patches a bit more descriptive name or add a commit message to them. Right now it isn't really obvious what they are doing. Also would upstream maybe accept feature flags for those instead?

Reply:
All the patches have only one purpose: remove the prompt change of the shell, because in NixOS usually the shell config is declarative and the related .bashrc, .zshrc and config.fish are read-only so the program cannot edit them as occurs in different Linux systems. Maybe we can just add a single comment on top of `patches` with a small description.

Reply:
I think patches usually are structured into chunks of logical changes. If the intention behind all the patches are the same, then they should ideally be collected into one single large patch, named with the intended effect. The comment would mostly be for further clarification.

Reply:
@h7x4 fixed.